bentoml/_internal/utils/cattr.py

```diff
@@ -28,5 +26,23 @@ def datetime_decoder(dt_like: str | datetime, _: t.Any) -> datetime:
         raise Exception(f"Unable to parse datetime from '{dt_like}'")
 
 
-bentoml_cattr.register_unstructure_hook(datetime, lambda dt: dt.isoformat())  # type: ignore
-bentoml_cattr.register_structure_hook(datetime, datetime_decoder)
+bentoml_cattr.register_structure_hook_factory(
+    attr.has,
+    lambda cls: make_dict_structure_fn(
+        cls,
+        bentoml_cattr,
+        _cattrs_forbid_extra_keys=getattr(cls, "__forbid_extra_keys__", False),
```

Comment:
I read that dunders (double leading and trailing underscore) should be reserved for python internals. Unlikely we will get a collision with this name but should we by consistent with naming and maybe use ALL_UPPERCASE or _single_leading_underscore?

Reply:
I think dunder usage in this case is fine, since this is only specific to a bentoml internal. The best way is to use metaclass, but then there is no need for that complexity.